infra/Pos/NtpCheck.hs

```diff
@@ -63,3 +69,14 @@ timeDifferenceWarnInterval = fromIntegral (Infra.ccTimeDifferenceWarnInterval in
 
 timeDifferenceWarnThreshold :: HasInfraConfiguration => Microsecond
 timeDifferenceWarnThreshold = fromIntegral (Infra.ccTimeDifferenceWarnThreshold infraConfiguration)
+
+type NtpStatusVar = TVar NtpStatus
+
+-- Helper to get ntp status
+mkNtpStatusVar :: ( NtpCheckMonad m , Mockables m [ CurrentTime, Delay] )
+    => m NtpStatusVar
+mkNtpStatusVar = do
+    status <- newTVarIO NtpSyncOk
```

Comment:
I think you should use an MVar and create it empty. The reason is that if the front-end makes a request before the first NTP check is performed, we'd probably want to wait for the check rather than say that everything is OK (when it's possibly not).

I don't know what the desired behavior is with regards to this, so I'm going to approve, but keep this suggestion in mind. And if the current behavior is intended, perhaps add a clarifying comment.